### PR TITLE
Fixes #17282: Documentation for group properties

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/17282

### CODE REVIEW

File: src/reference/modules/reference/pages/variables.adoc [hunk 1]
```diff
@@ -1,93 +1,404 @@
...
 
-== General
+== Generalities
 
-In Rudder, you can define and use variables via the technique editor or the differents pre-built techniques.
+Rudder has a notion of `variables`. A `variable` is something with a `name` that references a `value`, and that `value` can be accessed with appropriate syntax around its `name`. We call accessing the value `variable expansion` in honor to shell, guarding totem of devops glue. 
 
-All variables are defined under a `prefix` (scope), so to reference a variable you will always need its `prefix` and its `name`, separated via a `.` char.
-To call a variable in Rudder we use `${...}` brackets syntax as described below:
+In Rudder, `variable name` are namespaced with a dotted convention. `Values` can be maonly of two types: `string` and `json`, ie dictionnary-like recursice structures. Rudder only supports `UTF-8` for `varibles` `names` and `values`.
```
Comment (amousset) [inline src/reference/modules/reference/pages/variables.adoc:7]:
```suggestion
In Rudder, `variable name` are namespaced with a dotted convention. `Values` can be mainly of two types: `string` and `json`, ie dictionnary-like recursive structures. Rudder only supports `UTF-8` for `variables` `names` and `values`.
```

File: src/reference/modules/reference/pages/variables.adoc [hunk 2]
```diff
@@ -1,93 +1,404 @@
...
+Under that broad notion of `variable`, we need to take appart the two main action around them:
+
+- how one defines them? 
+- and how one uses them?
+
+In each case, we have subacase directed by use. 
+
+Variables definition covers two big aspects of knowledge:
+
+- configuration data, known by humans when policies are defined. That knowledge need to be centralized, categorized, hierarchized and retrieved based on what node
```
Comment (amousset) [inline src/reference/modules/reference/pages/variables.adoc:18]:
```suggestion
- configuration data, known by humans when policies are defined. That knowledge needs to be centralized, categorized, hierarchized and retrieved based on what node
```

File: src/reference/modules/reference/pages/variables.adoc [hunk 3]
```diff
@@ -1,93 +1,404 @@
...
+ policies will be used, 
+- dynamic or contextualized information about infrastructure provided by environment tooling, be it Rudder itsefl, facts about environment, or resources available only when the policy is checked.
```
Comment (amousset) [inline src/reference/modules/reference/pages/variables.adoc:20]:
```suggestion
- dynamic or contextualized information about infrastructure provided by environment tooling, be it Rudder itself, facts about environment, or resources available only when the policy is checked. 
```

File: src/reference/modules/reference/pages/variables.adoc (hunk 1 above)
Comment (amousset) [reply src/reference/modules/reference/pages/variables.adoc:7]: Actually I think variable names need to be ASCII

Comment (fanf) [reply src/reference/modules/reference/pages/variables.adoc:7]: the need for cfengine, but they are escaped by agent during generation. Is there an other reason to do so?

Comment (amousset) [reply src/reference/modules/reference/pages/variables.adoc]: They are put as is in properties.json, but when trying to use a variable name containing an emoji in a directive parameter I get:

> ⇨ Accumulated: Unexpected: Error with parameters expansion for node 'server.rudder.local' (root); cause was: Accumulated: When processing directive 'File content'; cause was: Accumulated: On variable 'GENERIC_FILE_CONTENT_PAYLOAD':; cause was: Accumulated: Unexpected: Error when parsing value "${node.properties[emo👋ji]}", error message is:

We can keep UTF-8, but what I fear is that a lot of things will break in unexpected ways.

Comment (fanf) [reply src/reference/modules/reference/pages/variables.adoc]: OK, what shoud we authorize? `[\-_0-9a-zA-Z]`? `~` ? All ascii without control characters?

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by amousset** [comment]
There are a lot of typos, could you check the file with a spell checker? It will be easier than reporting them as review comments.

File: src/reference/modules/reference/pages/variables.adoc [hunk 4]
```diff
@@ -1,337 +1,1096 @@
...
+
+===== Property syntax
+
+[cols="3", options="header"]
+|===
+|Use case
+|name=datacenter, value="Paris"
+|name=datacenter, value={"dns": "1.1.1.1"}
+
+|Technique CFEngine
```
Comment (amousset) [inline src/reference/modules/reference/pages/variables.adoc:218]: Classic technique

File: src/reference/modules/reference/pages/variables.adoc [hunk 5]
```diff
@@ -1,337 +1,1096 @@
...
+|${node.properties[datacenter]}
+|${node.properties[datacenter][dns]}
+
+|Technique DSC
```
Comment (amousset) [inline src/reference/modules/reference/pages/variables.adoc:222]: DSC technique

File: src/reference/modules/reference/pages/variables.adoc [hunk 6]
```diff
@@ -1,337 +1,1096 @@
...
+|Unqualified hostname
+
+|sys.host
+|Node's hostname (according to the kernel)
+
+|sys.domain
+|Node's domain as discovered by the agent
+|===
+
+There are also more variables available, all documented in https://docs.cfengine.com/docs/3.10/reference-special-variables-sys.html[this page].
```
Comment (amousset) [inline src/reference/modules/reference/pages/variables.adoc:509]:
```suggestion
There are also more variables available, all documented in https://docs.cfengine.com/docs/3.15/reference-special-variables-sys.html[this page].
```

File: src/reference/modules/reference/pages/variables.adoc [hunk 7]
```diff
@@ -1,8 +1,727 @@
...
+===== Property syntax
+
+[cols="3", options="header"]
+|===
+|Use case
+|name=datacenter, value="Paris"
+|name=datacenter, value={"dns": "1.1.1.1"}
+
+|Technique CFEngine
+|${node.properties[datacenter]}
```
Comment (amousset) [inline src/reference/modules/reference/pages/variables.adoc:219]:
```suggestion
| `${node.properties[datacenter]}`
```

**Comment by fanf** [comment]
PR rebased

File: src/reference/modules/reference/pages/variables.adoc [hunk 8]
```diff
@@ -1,13 +1,732 @@
...
+
+Rudder provides a simple way to add common and reusable variables in either plain directives, or techniques created using the technique editor: the parameters.
+
+image::rudder-parameters.png[Parameters]
+
+
+==== Group properties
+
+
+Group properties can be managed by using https://docs.rudder.io/api/#operation/updateGroup[goup's API] or they can be found in the *properties* tab of each group in Rudder UI.
```
Comment (amousset) [inline src/reference/modules/reference/pages/variables.adoc:146]: group

File: src/reference/modules/reference/pages/variables.adoc [hunk 9]
```diff
@@ -1,13 +1,732 @@
...
+
+====== Variable from files
+
+Rudder can natively parse JSON, YAML and CSV files to transform them into JSON values.
+
+====== Variable from `Augeas`
+
+http://augeas.net[Augeas] is standard tool for reading and editing configuration files by providing a tree-representation of their native format. It's really useful to clean the mess in that domain. With it, you can easily and safely transform almost any configuration file into a Rudder JSON variable. 
+
+====== Variable from `OSQuery`
```
Comment (amousset) [inline src/reference/modules/reference/pages/variables.adoc:641]:
```suggestion
====== Variable from `osquery`
```

**Comment by amousset** [comment]
Merging, I'll make another PR to update some little details

**Pull request merged**